rest/rest_collection.cc

```diff
@@ -670,11 +670,12 @@ void
 BackgroundTaskBase::
 cancel()
 {
-    bool wasCancelled = this->cancelled.exchange(true);
```

Comment:
I've noticed indeed that it was sensitive.   This is why I added a few assert.  I am 100% for the stress test.  Doing it now!